osc-server/src/main/java/org/osc/core/broker/util/crypto/X509TrustManagerFactory.java

```diff
@@ -235,17 +287,123 @@ public void addEntry(File file) throws Exception {
     }
 
     @Override
-    public void addEntry(X509Certificate certificate, String newAlias) throws Exception {
-        if (fingerprintNotExist(getSha1Fingerprint(certificate))) {
-            this.keyStore.setCertificateEntry(newAlias, certificate);
-            try (FileOutputStream outputStream = new FileOutputStream(TRUSTSTORE_FILE)) {
-                this.keyStore.store(outputStream, getTruststorePassword());
+    public void replaceInternalCertificate(File zipFile, boolean doReboot) throws Exception {
+        // Disabling reboot for testing.
+
+        if (zipFile == null) {
+            throw new Exception(EXCEPTION_KEYPAIR_ZIP_MALFORMED);
+        }
+
+        Map<String, File> files = unzipFilePair(zipFile);
+        File pKeyFile = files.get(KEY);
+        File chainFile = files.get(CERT_CHAIN);
+
+        FileUtils.deleteQuietly(zipFile);
```

Comment:
why delete? is it for cleanup or security reason? as VM is a protected load. Adding comments will give clarity on the code.

Reply:
Comment added. These files are not needed after the replace method call.